pbft/pbft.cpp

```diff
@@ -262,43 +262,42 @@ pbft::setup_request_operation(const bzn::encoded_message& request, const request
 }
 
 void
-pbft::handle_request(const pbft_request& msg, const bzn::json_message& original_msg, const std::shared_ptr<session_base>& session)
+pbft::handle_request(const bzn_envelope& request_env, const std::shared_ptr<session_base>& session)
 {
     if (!this->is_primary())
     {
-        LOG(info) << "Forwarding request to primary: " << original_msg.toStyledString();
-        this->node->send_message(bzn::make_endpoint(this->get_primary()), std::make_shared<bzn::json_message>(original_msg));
+        LOG(info) << "Forwarding request to primary";
+        this->node->send_message(bzn::make_endpoint(this->get_primary()), std::make_shared<bzn_envelope>(request_env));
         return;
     }
 
-    if (msg.timestamp() < (this->now() - MAX_REQUEST_AGE_MS) || msg.timestamp() > (this->now() + MAX_REQUEST_AGE_MS))
+    if (request_env.timestamp() < (this->now() - MAX_REQUEST_AGE_MS) || request_env.timestamp() > (this->now() + MAX_REQUEST_AGE_MS))
     {
         // TODO: send error message to client
-        LOG(info) << "Rejecting request because it is outside allowable timestamp range: "
-            << original_msg.toStyledString();
+        LOG(info) << "Rejecting request because it is outside allowable timestamp range: " << request_env.ShortDebugString();
         return;
     }
 
-    auto smsg = original_msg.toStyledString();
-    auto hash = this->crypto->hash(smsg);
+    auto hash = this->crypto->hash(request_env);
 
     // keep track of what requests we've seen based on timestamp and only send preprepares once
-    if (this->already_seen_request(msg, hash))
+    if (this->already_seen_request(request_env, hash))
     {
         // TODO: send error message to client
-        LOG(info) << "Rejecting duplicate request: " << original_msg.toStyledString();
+        LOG(info) << "Rejecting duplicate request: " << request_env.ShortDebugString();
         return;
     }
-    this->saw_request(msg, hash);
+    this->saw_request(request_env, hash);
 
-    auto op = setup_request_operation(smsg, hash, session);
+    //TODO: keep track of what requests we've seen based on timestamp and only send preprepares once - KEP-329
+    auto op = setup_request_operation(request_env, session);
```

Comment:
I had previously moved the hashing out of setup_request_operation to avoid hashing the request twice. If you prefer it this way that's fine but I wasn't sure if crypto->hash() is an expensive operation and needed to be minimized.

Reply:
fair enough; I had thought it was there only for the determinism concern.